cutlery/_wordpiece.pyx

```diff
@@ -37,18 +42,112 @@ cdef class WordPieceProcessor:
             token_ids.append(match.piece_id)
             token_pieces.append(f"{prefix}{token[:match.prefix_len]}")
             token = token[match.prefix_len:]
-            pieces = &self.continuation_pieces
+            pieces = &self.continuation_piece_to_id
             prefix = "##"
         return token_ids, token_pieces
 
+    def decode(self, pieces: Iterable[int], *, unk_token: str = "<unk>") -> str:
+        """
+        Decode piece identifiers into string. Invalid piece identifiers
+        are replaced with the `unk_token` string.
+
+            ids (Iterable[int]): Piece IDs.
+            unk_token (str): Marker for unknown piece IDs.
+            RETURNS (str): Decoded string.
+        """
+        token_pieces = [self.piece_id_to_str(id, unk_token=unk_token) for id in pieces]
```

Comment:
If an unknown identifier is passed in, we should raise an exception. That seems like a serious error (eg. using an incorrect vocab with a model) and should fail noisily.

Reply:
Fixed.